storage/storage.js

```diff
@@ -77,6 +59,39 @@ class Storage {
 			this.isReady = false;
 		});
 	}
+
+	/**
+	 * Register an entity by initializing its object.
+	 * It would be accessible through `storage.entities.[identifier]
+	 *
+	 * @param {string} identifier - Identifier used to access the object from stoage.entities.* namespace
+	 * @param {BaseEntity} Entity - A class constructor extended from BaseEntity
+	 * @param {Object} [options]
+	 * @param {Boolean} [options.replaceExisting] - Replace the existing entity
+	 * @param {Array.<*>} [options.initParams] - Extra parameters to pass to initialization of entity
+	 * @returns {BaseEntity}
+	 */
+	register(identifier, Entity, options = {}) {
+		const existed = Object.keys(Storage.instance.entities).includes(identifier);
+
+		if (existed && !options.replaceExisting) {
+			throw new EntityRegistrationError(
+				`Entity ${identifier} already registered`
+			);
+		}
+
+		let args = [Storage.instance.adapter];
+
+		if (options.initParams) {
+			args = args.concat(options.initParams);
+		}
+
+		const entityObject = new Entity(...args);
```

Comment:
Entities only accepts `adapter` and `defaultFilters` as params.
Deconstructing `args` array will use whatever is passed by `options.initParams`

Reply:
Yes it is intended as any custom entity can have any parameters.